MiniApp/Progress+UIViewController.swift

```diff
@@ -0,0 +1,36 @@
+import UIKit
+
+extension UIViewController {
+
+    func showProgressIndicator() {
+        DispatchQueue.main.async {
+            let alert = UIAlertController(title: nil, message: "Please wait...", preferredStyle: .alert)
+            let loadingIndicator = UIActivityIndicatorView(frame: CGRect(x: 10, y: 5, width: 50, height: 50))
+            loadingIndicator.hidesWhenStopped = true
+            loadingIndicator.style = UIActivityIndicatorView.Style.gray
+            loadingIndicator.startAnimating()
+            alert.view.addSubview(loadingIndicator)
+            self.present(alert, animated: true, completion: nil)
+        }
+    }
+
+    func dismissProgressIndicator() {
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1, execute: {
```

Comment:
Yeah, in few scenarios, dismiss is called even before the progress indicator is displayed (esp when the segue is performed) that results in showing endless progress indicator.

Reply:
Ok but it sounds like a race condition solved by adding an arbitrary delay. I think it could be solved by moving show/display indicator to viewWillAppear/DidDisappear because viewDidLoad is too early for that

Reply:
Yeah, I tried that too. Before creating this PR I even tried to move them (without the arbitrary 
delay) in viewDidLoad/viewWillAppear like you said. It is throwing an warning like "Warning: Attempt to dismiss from view controller <> while a presentation or dismiss is in progress!" Progress indicator is not dismissed.

Reply:
I see. I remember that this flow is very fragile. I would prefer not to stick with this workaround but it might be not worth spending additional time for this sample app.

Reply:
I totally agree with that. I spent more time to figure out why it is not working the way it is supposed to be, then I came up with this workaround

Reply:
I totally agree with that. I had spent more time to figure out the actual issue behind this but unfortunately I could come up with only this workaround